api/v1/cluster_types.go

```diff
@@ -1559,6 +1561,63 @@ func (in ExternalCluster) GetServerName() string {
 	return in.Name
 }
 
+// EnsureOption represents whether we should enforce the presence or absence of
+// a Role in a PostgreSQL instance
+type EnsureOption string
+
+// values taken by EnsureOption
+const (
+	EnsurePresent EnsureOption = "present"
+	EnsureAbsent  EnsureOption = "absent"
+)
+
+// ManagedConfiguration represents the portions of PostgreSQL that are managed
+// by the instance manager
+type ManagedConfiguration struct {
+	Roles []RoleConfiguration `json:"roles,omitempty"`
+}
+
+// RoleConfiguration is the representation, in Kubernetes, of a PostgreSQL role
+// with the additional field Ensure specifying whether to ensure the presence or
+// absence of the role in the database
+//
+// The defaults of the CREATE ROLE command are applied
+// Reference: https://www.postgresql.org/docs/current/sql-createrole.html
+type RoleConfiguration struct {
+	Name    string `json:"name"`
+	Comment string `json:"comment,omitempty"`
+
+	// ensure defaults to "present"
+	// +kubebuilder:default:="present"
+	Ensure EnsureOption `json:"ensure,omitempty"`
+
+	PasswordSecret *PasswordConfiguration `json:"passwordSecret,omitempty"`
+	Superuser      bool                   `json:"superuser,omitempty"`
+	CreateDB       bool                   `json:"createdb,omitempty"`
+	CreateRole     bool                   `json:"createrole,omitempty"`
+
+	// inherit defaults to true
+	// +kubebuilder:default:=true
+	Inherit bool `json:"inherit,omitempty"` // IMPORTANT default is INHERIT
+
+	Login       bool `json:"login,omitempty"`
+	Replication bool `json:"replication,omitempty"`
+	BypassRLS   bool `json:"bypassrls,omitempty"` // Row-Level Security
+
+	// connection Limit defaults to `-1`
+	// +kubebuilder:default:=-1
+	ConnectionLimit int64 `json:"connectionLimit,omitempty"`
+
+	ValidUntil string   `json:"validUntil,omitempty"`
+	InRoles    []string `json:"in_role,omitempty"`
+	WithRoles  []string `json:"role,omitempty"`
```

Comment:
We should change this to be only `Roles` in place of `InRoles`. The `WithRoles` is redundant.

```suggestion
	Roles    []string `json:"role,omitempty"`
```

Reply:
As per discussion, deleted `WithRoles` and updated the JSON for InRoles `in_role` -> `inRoles`